### PR TITLE
docs: add client ping to example

### LINKED ISSUE #395: First example test does not fail if Docker is not running
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/dockertest/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/dockertest/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Network](https://www.ory.sh/) project.
- [X] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [ ] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe the bug

;On MacOS 13.0.1 with Docker Desktop 4.15.0.

When using the code specified in the readme, the first stage of `TestMain` does not fail if Docker is not running, yet the failure message is labelled `Could not connect to Docker`. Here is the offending code

```
func TestMain(m *testing.M) {
	// uses a sensible default on windows (tcp/http) and linux/osx (socket)
	pool, err := dockertest.NewPool("")
	if err != nil {
		log.Fatalf("Could not connect to docker: %s", err)
	}
}
```
If taken in isolation I believe this should have a different error message. I also believe we should be explicit about whether Docker is running or not by adding a second check using `Client.Ping()`, something like this:

```
func TestMain(m *testing.M) {
	// uses a sensible default on windows (tcp/http) and linux/osx (socket)
	pool, err := dockertest.NewPool("")
	if err != nil {
		log.Fatalf("Could not construct pool: %s", err)
	}

	// uses pool to try to connect to Docker
	err = pool.Client.Ping()
	if err != nil {
		log.Fatalf("Could not connect to Docker: %s", err)
	}
}
```
This was we check whether Docker is running before we start running any other testing scenarios.

I am preparing a PR for your consideration on this.

### Reproducing the bug

1. Do not have Docker running, verify this by running some Docker command, `docker ps` for example, should return `Cannot connect to the Docker daemon at unix:///Users/smbm/.docker/run/docker.sock. Is the docker daemon running?`
2. Run this example test code, which is based on the example in the readme:
```
package main

import (
	"github.com/ory/dockertest/v3"
	"log"
	"testing"
)

func TestMain(m *testing.M) {
	pool, err := dockertest.NewPool("")
	if err != nil {
		log.Fatalf("Could not connect to Docker: %s", err)
	}
	log.Println(err)
	log.Println(pool)
}
```
3. See that `err` is printed as being `<nil>`

### Relevant log output

```shell
go test -v .
2022/12/06 21:14:09 <nil>
2022/12/06 21:14:09 &{0x140001c0370 0s}
ok  	myapp	0.183s
```

### Relevant configuration

_No response_

### Version

3.9.1

### On which operating system are you observing this issue?

macOS

### In which environment are you deploying?

None

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by smbm**
Updated Readme and Examples to be explicit about whether Docker is running or not by using `Client.Ping()`. 
Updated error returned if pool construction fails to refer to that specifically rather than `Could not connect to Docker` which never failed even if Docker was not running.
Updated the `dockertest_test.go` file to incorporate the same changes.

## Related Issue or Design Document

Fixes #395

## Checklist

- [X] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [X] I have read the [security policy](../security/policy).
- [X] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added the necessary documentation within the code base (if appropriate).

## Further comments

I have run the tests as best I can but on an M1 Mac I am unable to run the MySQL 5.7 and Mongo 3.3.12 tests due to mismatching architectures. If I comment these tests out everything passes, I realise this is suboptimal.

I have not made any changes to dockertest itself, only to the tests, readme and examples. If the maintainers feel that more testing is needed then I can see what I can do. I only have Arm hardware to hand at the moment though.

### CODE REVIEW

File: examples/BuildDockerfile.md [hunk 1]
```diff
@@ -11,7 +11,12 @@ FROM postgres:latest
 ```go
 pool, err := dockertest.NewPool("")
 if err != nil {
-	log.Fatalf("Could not connect to docker: %s", err)
+	log.Fatalf("Could not construct pool: %s", err)
+}
+
+err = pool.Client.Ping()
+if err != nil {
+    log.Fatalf("Could not connect to Docker: %s", err)
```
Comment (alnr) [inline examples/BuildDockerfile.md]: Indentation is with spaces instead of tabs here.

File: examples/Cassandra.md [hunk 2]
```diff
@@ -4,7 +4,12 @@ var err error
 
 pool, err := dockertest.NewPool("")
 if err != nil {
-    log.Fatalf("Could not connect to docker: %s", err)
+	log.Fatalf("Could not construct pool: %s", err)
+}
+
+err = pool.Client.Ping()
+if err != nil {
+    log.Fatalf("Could not connect to Docker: %s", err)
```
Comment (alnr) [inline examples/Cassandra.md:12]: same here

File: examples/CockroachDB.md [hunk 3]
```diff
@@ -3,7 +3,12 @@ var db *sql.DB
 var err error
 pool, err := dockertest.NewPool("")
 if err != nil {
-    log.Fatalf("Could not connect to docker: %s", err)
+	log.Fatalf("Could not construct pool: %s", err)
+}
+
+err = pool.Client.Ping()
+if err != nil {
+    log.Fatalf("Could not connect to Docker: %s", err)
```
Comment (alnr) [inline examples/CockroachDB.md:11]: same here

**Review by alnr — CHANGES_REQUESTED** [review]
Thank you for this great contribution! Would you mind checking the example markdown files for consistent indentation with tabs? I marked a couple of occasions.

Otherwise this looks good to merge 👍

**Comment by smbm** [comment]
> Thank you for this great contribution! Would you mind checking the example markdown files for consistent indentation with tabs? I marked a couple of occasions.
> 
> Otherwise this looks good to merge 👍

Thanks for the response! I have checked each of the example .md files and corrected the indenting to be consistent with the rest of that file. If you'd prefer I could fix the indenting of all lines in each file to be tabs or spaces.

**Review by alnr — APPROVED** [review]

**Review by zepatrik — APPROVED** [review]

**Comment by vinckr** [comment]
Hello @smbm 
Congrats on merging your first PR in Ory 🎉 !
Your contribution will soon be helping secure millions of identities around the globe 🌏. 
As a small token of appreciation we send our first time contributors a gift package to welcome them to the community. 
[Please drop me an email and I will forward you the form to claim your Ory swag!](mailto:vincent@ory.sh)

**Pull request merged**